cmd/kube-controllers/main.go

```diff
@@ -121,8 +130,64 @@ func main() {
 	// If configured to do so, start an etcdv3 compaction.
 	startCompactor(ctx, config)
 
-	// Wait forever.
-	select {}
+	sigint := make(chan os.Signal, 1)
+	signal.Notify(sigint, os.Interrupt)
+
+	// Wait forever and perform healthchecks.
+	for {
+		select {
+		case <-time.After(10 * time.Second):
+			// Datastore HealthCheck
+			healthCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+			err = calicoClient.EnsureInitialized(healthCtx, "", "k8s")
+			if err != nil {
+				log.WithError(err).Errorf("Failed to verify datastore")
+				s.SetReady(
+					"CalicoDatastore",
+					false,
+					fmt.Sprintf("Error verifying datastore: %v", err),
+				)
+			} else {
+				s.SetReady("CalicoDatastore", true, "")
+			}
+			cancel()
+
+			// Kube-apiserver HealthCheck
+			client := http.Client{
+				Timeout: time.Duration(time.Second),
```

Comment:
Might want to bump this timeout to something like 5 seconds, just in case.

Should never happen, but bumping to 5 seconds won't hurt anyone either :)